oci/oci.go

```diff
@@ -435,29 +435,36 @@ func (r *Runtime) ExecSync(c *Container, command []string, timeout int64) (resp
 	}, nil
 }
 
-// StopContainer stops a container.
-func (r *Runtime) StopContainer(c *Container) error {
+// StopContainer stops a container. Timeout is given in seconds.
+func (r *Runtime) StopContainer(c *Container, timeout int64) error {
 	c.opLock.Lock()
 	defer c.opLock.Unlock()
 	if err := utils.ExecCmdWithStdStreams(os.Stdin, os.Stdout, os.Stderr, r.Path(c), "kill", c.name, "TERM"); err != nil {
 		return err
 	}
-	i := 0
-	for {
-		if i == 1000 {
-			err := unix.Kill(c.state.Pid, syscall.SIGKILL)
-			if err != nil && err != syscall.ESRCH {
-				return fmt.Errorf("failed to kill process: %v", err)
+	if timeout == -1 {
+		// default 2 seconds delay
```

Comment:
changed to `10` seconds as it's now in docker https://github.com/moby/moby/blob/master/container/container.go#L53 (not sure why you said "dockershim" though :confused:)